### PR TITLE
Capture ocr runtime errors

### PR DESCRIPTION
**Pull request by connorwstein**
The chainlink node can accept new job spec posts while its disconnected from an eth node, so it doesn't validate some of the parameters (e.g. a contract address in a spec). If there are errors between specifying a job and running it (i.e. contract self-destructs in between) we still want to capture those errors, yet there are not specific to a job task but issues with the spec itself. Those errors are currently saved in the job_spec_errors table. 

We want to do something analogous for the v2 job pipeline, so I added a similar table: pipeline_spec_errors. The only current job type using the pipeline is ocr type jobs and to capture those spec errors, I put a function closure on the logger which has access to pipeline.ORM to save the errors.

### CODE REVIEW

File: .github/workflows/static.yml
Comment (j16r) [inline .github/workflows/static.yml:22]: Could you split this into a separate PR?

File: core/services/pipeline/runner.go
Comment (j16r) [inline core/services/pipeline/runner.go:25]: This name reveals the implementation, something like "RecordError" might be better.

File: core/services/offchainreporting/logger.go
Comment (j16r) [inline core/services/offchainreporting/logger.go:46]: Having a Logger#Error function with a side effect other than simply logging might be a bit surprising, could we move this into a different abstraction?

File: .github/workflows/static.yml
Comment (connorwstein) [reply .github/workflows/static.yml]: https://github.com/smartcontractkit/chainlink/pull/3568/files

File: core/services/offchainreporting/logger.go
Comment (connorwstein) [reply core/services/offchainreporting/logger.go:46]: yeah it did feel a little strange, however introducing a new abstraction would require changing the libocr interface as far as I can tell. Not sure we want to block on that cc @spooktheducks

File: core/services/offchainreporting/oracle.go
Comment (samsondav) [inline core/services/offchainreporting/oracle.go:119]: Can we reduce this query to only fetch the pipeline spec ID, since that's all we need?

File: core/services/pipeline/runner.go
Comment (samsondav) [inline core/services/pipeline/runner.go:140]: Can we modify this to take a context?

File: core/store/migrations/migration1604437959/migrate.go
Comment (samsondav) [inline core/store/migrations/migration1604437959/migrate.go:14]: Let's add BRIN indexes on created_at and updated_at.

**Review by samsondav — CHANGES_REQUESTED** [review]

File: core/store/migrations/migration1604437959/migrate.go
Comment (DeividasK) [reply core/store/migrations/migration1604437959/migrate.go:14]: Might this be a bit premature? Are we expecting to have lots of unique spec errors?

Comment (connorwstein) [reply core/store/migrations/migration1604437959/migrate.go:14]: Yeah without knowing anything about the load on this table yet, I think it is premature

File: core/services/offchainreporting/oracle.go
Comment (connorwstein) [reply core/services/offchainreporting/oracle.go:119]: Hmm I'm not sure I like the alternative better: 
```
	var pipelineSpecID []int32
	err = d.db.Table("jobs").
		Where("id = ?", spec.JobID()).
		Pluck("pipeline_spec_id", &pipelineSpecID).
		Error
	if err != nil {
		return nil, errors.Wrapf(err, "could not find job with ID %d", spec.JobID())
	}
        // awkwardly use pipelineSpecID[0] ?
```

Comment (connorwstein) [reply core/services/offchainreporting/oracle.go:119]: given its a 3 column table I think its actually better as is

Comment (samsondav) [reply core/services/offchainreporting/oracle.go:119]: Yup I agree

**Review by samsondav — DISMISSED** [review]
LGTM, don't forget to squash before merge

**Review by se3000 — APPROVED** [review]

**Pull request merged**
